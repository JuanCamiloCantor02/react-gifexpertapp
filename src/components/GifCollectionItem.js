import React from 'react'

export const GifCollectionItem = ({id, title, url}) => {
   
    
   
    return (
        <div className="card animate__animated animate__backInUp">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}
