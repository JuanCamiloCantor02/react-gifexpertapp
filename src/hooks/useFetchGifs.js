
import { useEffect, useState } from 'react'
import {getCollection} from '../helpers/getGifsCollection'


export const useFetchGifs = (category) => {
    
    const [state, setState] = useState({
        data : [],
        loading : true
    });


    useEffect(() =>{

        getCollection(category)
            .then(imgs =>{
                
                setTimeout(() => {
                    
                    
                    setState({
                        data: imgs,
                        loading: false
                    });


                }, 2000);
                
                    
            })

    }, [category])

    /* setTimeout(() => {
        setState({
            data: [1,2,3,4,5,6,7,8,9],
            loading: false
        })
    }, 3000); */

    return state;

}


