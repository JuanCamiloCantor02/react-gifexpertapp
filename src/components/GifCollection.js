import React/* , { useEffect, useState } */ from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
/* import { getCollection } from '../helpers/getGifs'; */
import { GifCollectionItem } from './GifCollectionItem';

export const GifCollection = ({category}) => {
   
    /* const [images, setImages] = useState([]);
    
    useEffect(() => {
        getCollection(category)
            .then(setImages);
    }, [category]) */

    const {data:images, loading} = useFetchGifs(category);

    


    return (
        <>
            <h3 className="animate__animated animate__pulse">{category}</h3>

            {loading && <p>loading...</p>}
            
            <div className="card-collection">
                
                    {
                        images.map( img =>(
                        <GifCollectionItem
                            key={img.id} 
                            {...img} 
                        />
                        ))
                    }
                
            </div>
        </>
    )
}
