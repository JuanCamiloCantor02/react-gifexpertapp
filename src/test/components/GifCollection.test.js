import React from "react"
import "@testing-library/jest-dom";
import { GifCollection } from "../../components/GifCollection";
import {shallow} from "enzyme";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock("../../hooks/useFetchGifs")

describe('pruebas en <GifCollection/>', () => {
    
    const category = "goku"
    
    test('debe mostrarse correctamente ', () => {
        
        useFetchGifs.mockReturnValue({
            data : [],
            loading : true
        });
        
        const wrapper = shallow(<GifCollection category= {category} />);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: "ABC",
            url: "https://123456/456564.jpg",
            title: "COSA"
        },
        {
            id: "cde",
            url: "https://dsadsadsa/dsadasdas.jpg",
            title: "Lechona"
        },
        {
            id: "fgj",
            url: "https://456das5644das/dsa54d64as5465.jpg",
            title: "Pan"
        }]

        useFetchGifs.mockReturnValue({
            data : gifs,
            loading : false
        });
        
        const wrapper = shallow(<GifCollection category= {category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifCollectionItem").length).toBe(gifs.length);

    })
    
    

})
