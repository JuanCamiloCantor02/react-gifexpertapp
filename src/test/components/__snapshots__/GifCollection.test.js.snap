// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifCollection/> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    goku
  </h3>
  <div
    className="card-collection"
  >
    <GifCollectionItem
      id="ABC"
      key="ABC"
      title="COSA"
      url="https://123456/456564.jpg"
    />
    <GifCollectionItem
      id="cde"
      key="cde"
      title="Lechona"
      url="https://dsadsadsa/dsadasdas.jpg"
    />
    <GifCollectionItem
      id="fgj"
      key="fgj"
      title="Pan"
      url="https://456das5644das/dsa54d64as5465.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas en <GifCollection/> debe mostrarse correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    goku
  </h3>
  <p>
    loading...
  </p>
  <div
    className="card-collection"
  />
</Fragment>
`;
